custom/src/main/java/org/kiwix/kiwixmobile/custom/download/CustomDownloadFragment.kt

```diff
@@ -19,35 +19,42 @@
 package org.kiwix.kiwixmobile.custom.download
 
 import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
 import io.reactivex.disposables.CompositeDisposable
-import kotlinx.android.synthetic.main.activity_custom_download.cd_view_animator
+import kotlinx.android.synthetic.main.fragment_custom_download.cd_view_animator
 import kotlinx.android.synthetic.main.layout_custom_download_error.cd_error_text
 import kotlinx.android.synthetic.main.layout_custom_download_error.cd_retry_button
 import kotlinx.android.synthetic.main.layout_custom_download_in_progress.cd_download_state
 import kotlinx.android.synthetic.main.layout_custom_download_in_progress.cd_eta
 import kotlinx.android.synthetic.main.layout_custom_download_in_progress.cd_progress
 import kotlinx.android.synthetic.main.layout_custom_download_required.cd_download_button
 import org.kiwix.kiwixmobile.core.base.BaseActivity
-import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.base.BaseFragment
+import org.kiwix.kiwixmobile.core.base.FragmentActivityExtensions
 import org.kiwix.kiwixmobile.core.downloader.model.DownloadItem
-import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.viewModel
 import org.kiwix.kiwixmobile.core.extensions.setDistinctDisplayedChild
+import org.kiwix.kiwixmobile.core.extensions.viewModel
+import org.kiwix.kiwixmobile.core.main.CoreMainActivity
 import org.kiwix.kiwixmobile.custom.R
 import org.kiwix.kiwixmobile.custom.customActivityComponent
 import org.kiwix.kiwixmobile.custom.download.Action.ClickedDownload
 import org.kiwix.kiwixmobile.custom.download.Action.ClickedRetry
+import org.kiwix.kiwixmobile.custom.download.Action.PressedBack
 import org.kiwix.kiwixmobile.custom.download.State.DownloadComplete
 import org.kiwix.kiwixmobile.custom.download.State.DownloadFailed
 import org.kiwix.kiwixmobile.custom.download.State.DownloadInProgress
 import org.kiwix.kiwixmobile.custom.download.State.DownloadRequired
 import javax.inject.Inject
 
-class CustomDownloadActivity : BaseActivity() {
-  override fun injection(coreComponent: CoreComponent) {
-    customActivityComponent.inject(this)
-  }
+var pressBackCallback = 0
```

Comment:
we should not need this variable, arbitrarily accessing top level variables is a surefire path to coupling

Reply:
Removed it